contracts/protocol/libs/EIP712.sol

```diff
@@ -63,7 +61,7 @@ contract EIP712 is SignatureErrors {
      * @return the domain separator
      */
     function getDomainSeparator() internal view returns (bytes32) {
-        if (address(this) == FERMION_PROTOCOL_ADDRESS && block.chainid == CHAIN_ID_CACHED) {
```

Comment:
This part of the remediation is incorrect. At this point we still need to compare if the current `block.chainId` matches the chainID at the deployment time. If we don't do that, we'll always return `DOMAIN_SEPARATOR_CACHED` which was calculated using the old chain id.  

Only the `CHAIN_ID_CACHED` in `buildDomainSeparator` had to be replaced.